xla/myplugin/myplugin_cpp_pjrt.h

```diff
@@ -0,0 +1,160 @@
+#ifndef MYPLUGIN_PJRT_CPP_H
+#define MYPLUGIN_PJRT_CPP_H
+
+#include <vector>
+
+#include "xla/pjrt/pjrt_client.h"
+
+namespace myplugin_pjrt {
+  using namespace xla;
+
+  class MypluginPjrtClient : public xla::PjRtClient {
+    public:
+      MypluginPjrtClient() {};
+      ~MypluginPjrtClient() override {};
+      absl::string_view platform_name() const override;
+      int process_index() const override;
+      int device_count() const override { return 42; }
+      int addressable_device_count() const override { return 43; }
+      absl::Span<xla::PjRtDevice* const> devices() const override { return devices_; }
+      absl::Span<xla::PjRtDevice* const> addressable_devices() const override { return devices_; }
+
+      absl::StatusOr<PjRtDevice*> LookupAddressableDevice(
+              PjRtLocalDeviceId local_device_id) const override {
```

Comment:
There's a ton of unimplemented.. should we just make these virtual methods with default "unimplemented" impls? Can still be an abstract base class by requiring a few methods be implemented?

Reply:
Indeed, we should! Half of them are already - https://github.com/openxla/xla/blob/main/xla/pjrt/pjrt_client.h#L575

That was a follow up but it's needed now to even compile.

Reply:
Done at HEAD now.